apps/central-scan/frontend/src/components/toggle_test_mode_button.test.tsx

```diff
@@ -79,7 +79,45 @@ test('calls the callback on confirmation', () => {
   expect(toggleTestMode).toHaveBeenCalled();
 });
 
-test('shows a modal when toggling to live mode', () => {
+test('toggle modal shows "official ballot mode" when toggling away from test ballot mode', () => {
+  const toggleTestMode = jest.fn();
+  const { getByText } = render(
+    <ToggleTestModeButton
+      canUnconfigure
+      isTestMode
+      isTogglingTestMode={false}
+      toggleTestMode={toggleTestMode}
+    />
+  );
+
+  // Click the button.
+  fireEvent.click(getByText('Toggle to Official Ballot Mode'));
+
+  getByText(
+    'Toggling to Official Ballot Mode will zero out your scanned ballots. Are you sure?'
+  );
```

Comment:
Collectively we've been trending toward a couple recommended React testing library patterns:

- use `userEvent` rather than `fireEvent`
- use queries bound to the screen (e.g. `screen.getByText`) rather than queries bound to a specific render

These both simulate human interaction better and test better for accessibility. I've been meaning to write a doc summarizing these sorts of things.

It's not high priority - we're not putting any effort into old tests - but when I write new tests I try to follow these patterns. It's so common we're using the old pattern I don't think it's worth you making a change here unless you feel so inspired.